File: tests/js/test_mfdb_validator.js
/**
 * Library:     test_mfdb_validator.js
 * Jurisdiction: ["JAVASCRIPT", "CORE_COMMAND"]
 * Status:      OFFICIAL — Core-Command/Lib (v1.1)
 * Author:      Elton Boehnen
 * Version:     1.3 OFFICIAL
 * Date:        2026-05-01
 * Description: Core-Command library component.
 */
const {
  mfdb_validator_validate_manifest,
} = require('../../js/lib_mfdb_validator.js');

const {
  mfdb_core_create_database,
} = require('../../js/lib_mfdb_core.js');

const fs = require('fs');
const path = require('path');
const assert = require('assert');

function runTests() {
  console.log('Running JavaScript MFDB Validator Tests...');
  let passed = 0;
  let failed = 0;

  const testDir = path.join(__dirname, 'tmp_mfdb_val');
  if (!fs.existsSync(testDir)) fs.mkdirSync(testDir);

  const entities = [
    {
      name: 'User',
      fields: [{ name: 'id', type: 'string' }, { name: 'name', type: 'string' }],
      primary_key: 'id'
    }
  ];

  function test(name, fn) {
    try {
      fn();
      passed++;
    } catch (err) {
      failed++;
      console.error(`[FAIL] ${name}: ${err.message}`);
    }
  }

  test('Manifest Validation', () => {
    const manifestPath = mfdb_core_create_database(testDir, 'TestDB', entities, 'A test db');
    assert.ok(mfdb_validator_validate_manifest(manifestPath));
  });

  // Cleanup
  try {
    const rm = (p) => {
      if (fs.statSync(p).isDirectory()) {
        fs.readdirSync(p).forEach(f => rm(path.join(p, f)));
        fs.rmdirSync(p);
      } else {
        fs.unlinkSync(p);
      }
    };
    rm(testDir);
  } catch (e) {}

  console.log(`---------------------------------------`);
  console.log(`JS MFDB Validator Results: ${passed} passed, ${failed} failed`);
  if (failed > 0) process.exit(1);
}

runTests();

File: tests/py/test_mfdb_validator.py
"""
Library:     test_mfdb_validator.py
Jurisdiction: ["PYTHON", "CORE_COMMAND"]
Status:      OFFICIAL — Core-Command/Lib (v1.1)
Author:      Elton Boehnen
Version:     1.3 OFFICIAL
Date:        2026-05-01
Description: Core-Command library component.
"""
import os
import sys
import unittest
import tempfile
import shutil

LIB_PY_DIR = "/storage/7B30-0E0B/Core-Command/Lib/py"
if LIB_PY_DIR not in sys.path:
    sys.path.insert(0, LIB_PY_DIR)

from lib_mfdb_core import mfdb_core_create_database
from lib_mfdb_validator import (
    mfdb_validator_validate_manifest,
    mfdb_validator_validate_entity_file,
    mfdb_validator_validate_database,
    MFDBValidationError
)

class TestMFDBValidator(unittest.TestCase):
    def setUp(self):
        self.test_dir = tempfile.mkdtemp()
        self.entities = [
            {
                "name": "User",
                "fields": [{"name": "id", "type": "string"}, {"name": "name", "type": "string"}],
                "primary_key": "id"
            }
        ]
        self.manifest_path = mfdb_core_create_database(
            self.test_dir, "TestDB", self.entities, "A test database"
        )

    def tearDown(self):
        shutil.rmtree(self.test_dir)

    def test_manifest_validation(self):
        # Should not raise
        self.assertTrue(mfdb_validator_validate_manifest(self.manifest_path))

    def test_entity_validation(self):
        user_path = os.path.join(self.test_dir, "data", "user.bejson")
        # Should not raise
        self.assertTrue(mfdb_validator_validate_entity_file(user_path, self.manifest_path))

    def test_database_validation(self):
        # Should not raise
        self.assertTrue(mfdb_validator_validate_database(self.manifest_path))

if __name__ == "__main__":
    unittest.main()
